lib/vsc/filesystem/oceanstor.py

```diff
@@ -151,11 +168,216 @@ def __init__(self, url, username=None, password=None):
         # Get token for this session
         self.session.client.get_x_auth_token(password)
 
-    def list_filesystems(self):
+    def list_storage_pools(self, update=False):
+        """
+        List all storage pools (equivalent to devices in GPFS)
+
+        Set self.storagepools to a convenient dict structure of the returned dict
+        where the key is the storagePoolName, the value is a dict with keys:
+        - storagePoolId
+        - storagePoolName
+        - totalCapacity
+        - reductionInvolvedCapacity
+        - allocatedCapacity
+        - usedCapacity
+        - freeCapacityRate
+        - usedCapacityRate
+        - status
+        - progress
+        - securityLevel
+        - redundancyPolicy
+        - numParityUnits
+        - numFaultTolerance
+        - compressionAlgorithm
+        - deduplicationSaved
+        - compressionSaved
+        - deduplicationRatio
+        - compressionRatio
+        - dataReductionRatio
+        - encryptType
+        - supportEncryptForMainStorageMedia
+        """
+        if not update and self.storagepools:
+            return self.storagepools
+
+        # Request storage pools
+        _, response = self.session.data_service.storagepool.get()
+        storage_pools = [sp['storagePoolName'] for sp in response['storagePools']]
+        self.log.debug("Storage pools in OceanStor: %s", ', '.join(storage_pools))
```

Comment:
can't you first do the bit below with the dict and then just call it with `.keys()`?

Reply:
done in https://github.com/vub-hpc/vsc-filesystem-oceanstor/pull/2/commits/d06dbc4fec54108aa66171de11b247be1f380f70